File: project/security/src/main/java/org/tby/fourdk/security/Authorization.java
package org.tby.fourdk.security;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class Authorization {

    public final Boolean isAuthorized;

    public final List<Permission> permissions;

    public Authorization(Boolean isAuthorized, Collection<Permission> permissions) {
        this.isAuthorized = isAuthorized;
        this.permissions = new ArrayList<>(permissions);
    }

}

File: rust/project/security-rust/src/dddk/security/authorization.rs
use crate::dddk::security::permission::Permission;

pub struct Authorization {
    is_authorized: bool,
    permissions: Vec<Permission>,
}

impl Authorization {
    pub fn new(is_authorized: bool, permissions: Vec<Permission>) -> Authorization {
        Authorization {
            is_authorized,
            permissions,
        }
    }

    pub fn is_authorized(&self) -> bool {
        self.is_authorized
    }

    pub fn get_permissions(&self) -> &Vec<Permission> {
        &self.permissions
    }
}